public/scripts/app.js

```diff
@@ -1,7 +1,151 @@
 console.log("Sanity Check: JS is working!");
 
+var $duckiesArray;
+var allDucks = [];
+var prof;
+
 $(document).ready(function(){
 
-// your code
+  $duckiesArray = $('#ducksTarget');
+
+  // collecting duck data:
+  $.ajax({
+    method: 'GET',
+    url: '/api/ducks',
+    success: handleSuccess,
+    //error: handleError
+  });
+
+  //  adding new duck data:
+  $('#newDuckForm').on('submit', function(e) {
+    e.preventDefault();
+    $.ajax({
+      method: 'POST',
+      url: '/api/ducks',
+      data: $(this).serialize(),
+      success: newDuckSuccess,
+      //error: newDuckError
+    });
+  });
+
+  //  deleting duck data:
+  $duckiesArray.on('click', '.deleteBtn', function() {
+
+    $.ajax({
+      method: 'DELETE',
+      url: '/api/ducks/'+$(this).attr('data-id'),
+      success: duckDeleteSuccess,
+      error: duckDeleteError
+    });
+  });
+
+  // get profile data:
+  $.ajax({
+    method: 'GET',
+    url: '/api/profile',
+    success: profileSuccess
+  });
 
 });
+
+
+
+// get duck formatted into HTML elements
+function getDuckHTML(d) {
+  let g;
+  `${d.gender}`.toUpperCase() === "F" ? g = ["She", "Her"] : g = ["He", "His"];
+  return
+  $('#ducksTarget').append(
+    `<div class="row>">
+      <div class="col-xs-10">
+        <li>${d.name} belongs to ${d.bff}. ${d.description} ${g[1]} favorite quote is '${d.favQuote}'</li>
+      </div>
+      <div class="col-xs-2">
+        <button type="button" name="button" class="deleteBtn btn btn-danger pull-right" data-id=${d._id}>Delete Duck</button>
+      </div>
+    </div><br/>`);
+};
+
+// get all ducks in given array formatted into HTML elements
+function getAllDucksHTML(ducks) {
+
+  let result = ducks.forEach(function(d){
+    let g;
+    `${d.gender}`.toUpperCase() === "F" ? g = ["She", "Her"] : g = ["He", "His"];
```

Comment:
Change to one line and get rid of weird string-conversion:

`let g = (d.gender.toUpperCase() === "F" ? ["She", "Her"] : ["He", "His"]);`